table/tables/partition_test.go

```diff
@@ -729,3 +729,14 @@ func TestIssue31721(t *testing.T) {
 	tk.MustExec("insert into t_31721 values ('1')")
 	tk.MustExec("select * from t_31721 partition(p0, p1) where col1 != 2;")
 }
+
+func TestPruneMode(t *testing.T) {
+	store, _, clean := testkit.CreateMockStoreAndDomain(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("set @@tidb_partition_prune_mode = 'static'")
```

Comment:
Please add a check `tk.MustQuery("show warnings")` here to verify that no warning is reported.